### PR TITLE
Add converters and helpers (bs.deriving)

### PR DESCRIPTION
**Pull request by ryyppy**

### CODE REVIEW

File: pages/docs/manual/latest/generate-converters-accessors.mdx [hunk 1]
```diff
@@ -0,0 +1,565 @@
...
+
+</CodeTab>
+
+Variants constructors with payloads generate functions, payload-less constructors generate plain integers (the internal representation of variants).
+
+**Note**:
+- The generated accessors are lower-cased.
+- You can now use these helpers on the JavaScript side! But don't rely on their actual values please.
+- Doesn't work with polymorphic variants.
+
```
Comment (bobzhang) [inline pages/docs/manual/latest/generate-converters-accessors.mdx:57]: it does not work with poly var, the reason is that polyvar does not need it, the encoding of poly var is standardised

File: pages/docs/manual/latest/generate-converters-accessors.mdx [hunk 2]
```diff
@@ -0,0 +1,565 @@
...
+let opaqueValue = fruitToJs(Apple)
+
+@bs.module("myJSFruits") external jsKiwi: abs_fruit = "iSwearThisIsAKiwi"
+let kiwi = fruitFromJs(jsKiwi)
+
+let error = fruitFromJs(100) /* nope, can't take a random int */
+```
+
+## Generate Converters for JS String Enums and Polymorphic Variants
+
```
Comment (bobzhang) [inline pages/docs/manual/latest/generate-converters-accessors.mdx:277]: The best practice: use jsConverter to convert between int and rescript types for regular variants. For string converter, it is not needed any more in ReScript

**Review by bobzhang — COMMENTED** [review]
In general, this looks great!
I suggest we make a comment that user can use `deriving` directly

File: pages/docs/manual/latest/generate-converters-accessors.mdx (hunk 2 above)
Comment (ryyppy) [reply pages/docs/manual/latest/generate-converters-accessors.mdx:277]: @bobzhang I believe we are still missing polyvariant docs to properly reflect that best practise, but I will add a version notice similar to the JS object converter

File: pages/docs/manual/latest/generate-converters-accessors.mdx (hunk 1 above)
Comment (ryyppy) [reply pages/docs/manual/latest/generate-converters-accessors.mdx:57]: should i completely drop that line?

**Comment by ryyppy** [comment]
> I suggest we make a comment that user can use deriving directly
Which minimal version allows `@deriving` instead of `@bs.deriving`?

**Comment by bobzhang** [comment]
we can add a note that since 8.3, the bs. prefix is no longer needed

File: pages/docs/manual/latest/generate-converters-accessors.mdx (hunk 1 above)
Comment (ryyppy) [reply pages/docs/manual/latest/generate-converters-accessors.mdx:57]: dropped it

File: pages/docs/manual/latest/generate-converters-accessors.mdx (hunk 2 above)
Comment (ryyppy) [reply pages/docs/manual/latest/generate-converters-accessors.mdx:277]: added a note that since ReScript  >= 8.2 compiles polyvars to strings

**Comment by ryyppy** [comment]
@bobzhang i added a notice in the beginning that starting from 8.3, ppl can drop the `bs.` prefix. When we lift the docs version number to 8.4 (after adding the rest of the missing docs), I'd completely remove the `bs.` from all examples and turn around the notice that `bs.*` is still supported but deprecated

**Pull request merged**